library/lib/platforms/switch/switch_audio.cpp

```diff
@@ -51,18 +52,28 @@ SwitchAudioPlayer::SwitchAudioPlayer()
         return;
     }
 
-    // Mount qlaunch ROMFS for the BFSAR
-    Result result = romfsMountDataStorageFromProgram(QLAUNCH_PID, QLAUNCH_MOUNT_POINT);
-    if (!R_SUCCEEDED(result))
-    {
-        Logger::error("Unable to mount qlaunch ROMFS: {:#x}", result);
-
-        plsrPlayerExit();
-        return;
+    // Check if the program running is qlaunch
+    char bfsarPath[29];
+    u64 programId = 0;
+    svcGetInfo(&programId, InfoType_ProgramId, CUR_PROCESS_HANDLE, 0);
+    if (programId != QLAUNCH_PID) {
+        // Mount qlaunch ROMFS for the BFSAR
+        Result result = romfsMountDataStorageFromProgram(QLAUNCH_PID, QLAUNCH_MOUNT_POINT);
+        if (!R_SUCCEEDED(result))
+        {
+            Logger::error("Unable to mount qlaunch ROMFS: {:#x}", result);
+
+            plsrPlayerExit();
+            return;
+        }
+
+        sprintf(bfsarPath, "%s%s", QLAUNCH_MOUNT_POINT, BFSAR_PATH);
```

Comment:
Does the compiler warn if we add more characters to either defines, making the sprintf overflow the allocated 29 chars? If it doesn't warn, can you then change for snprintf, which does warn?

Reply:
:+1:  Yep it does:
```
/home/nichole/Projects/borealis/./library/lib/platforms/switch/switch_audio.cpp: In constructor 'brls::SwitchAudioPlayer::SwitchAudioPlayer()':
/home/nichole/Projects/borealis/./library/lib/platforms/switch/switch_audio.cpp:73:32: warning: '%s' directive writing 28 bytes into a region of size 23 [-Wformat-overflow=]
   73 |         sprintf(bfsarPath, "%s:%s", ROMFS_MOUNT_POINT, BFSAR_PATH);
      |                                ^~
/home/nichole/Projects/borealis/./library/lib/platforms/switch/switch_audio.cpp:73:16: note: 'sprintf' output 35 bytes into a destination of size 29
   73 |         sprintf(bfsarPath, "%s:%s", ROMFS_MOUNT_POINT, BFSAR_PATH);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/nichole/Projects/borealis/./library/lib/platforms/switch/switch_audio.cpp:70:32: warning: '%s' directive writing 28 bytes into a region of size 21 [-Wformat-overflow=]
   70 |         sprintf(bfsarPath, "%s:%s", QLAUNCH_MOUNT_POINT, BFSAR_PATH);
      |                                ^~
/home/nichole/Projects/borealis/./library/lib/platforms/switch/switch_audio.cpp:70:16: note: 'sprintf' output 37 bytes into a destination of size 29
   70 |         sprintf(bfsarPath, "%s:%s", QLAUNCH_MOUNT_POINT, BFSAR_PATH);
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```